readthedocs/projects/tasks/builds.py

```diff
@@ -316,6 +316,14 @@ def sigterm_received(*args, **kwargs):
 
         def sigint_received(*args, **kwargs):
             log.warning('SIGINT received. Canceling the build running.')
+
+            # Only allow to cancel the build if it's not already uploading the files.
+            # This is to protect our users to end up with half of the documentation uploaded.
+            # TODO: remove this condition once we implement "Atomic Uploads"
+            if self.data.build["state"] == BUILD_STATE_UPLOADING:
+                log.warning('Ignoring cancelling the build at "Uploading" state.')
```

Comment:
I think this will just end with the build getting KILL after INT, after an amount of time, but I guess that will take a bit longer?

Reply:
I'm not sure to understand your comment.

When the build process receive the INT signal, we are just ignoring it here. The build will continue uploading the files normally.

Reply:
Yes, but celery will send a KILL signal after the hard stop limit: https://docs.celeryq.dev/en/stable/userguide/configuration.html?highlight=settings#task-time-limit

Reply:
Yes. Hard stop limit is a different thing, tho. I opened a PR some time ago to work on that but it's not trivial: https://github.com/readthedocs/readthedocs.org/pull/9387